common_types/src/row/contiguous.rs

```diff
@@ -352,10 +349,15 @@ impl<'a, T: RowBuffer + 'a> ContiguousRowWriter<'a, T> {
                 let value_buf = (*next_string_offset as u32).to_ne_bytes();
                 Self::write_slice_to_offset(inner, offset, &value_buf);
 
-                // Encode length of string as a u32.
+                // Encode length of string as a varint.
                 ensure!(v.len() <= MAX_STRING_LEN, StringTooLong { len: v.len() });
-                let string_len = v.len() as u32;
-                Self::write_slice_to_offset(inner, next_string_offset, &string_len.to_ne_bytes());
+                let mut dst = [0; 4];
+                let value_buf = {
+                    let mut buf = &mut dst[..];
+                    encode_varint(v.len() as u64, &mut buf);
+                    &dst[..encoded_len_varint(v.len() as u64)]
```

Comment:
This block is similar to L379-L381, and how about define a free function called `encode_varint_to(&mut [u8]) -> &[u8]`.